src/ansys/dynamicreporting/core/utils/report_utils.py

```diff
@@ -51,20 +48,209 @@ def encode_url(s):
+        return None
+    return json_description
+
+
+def create_new_pil_image(pil_image):
+    """
+    Convert the existing PIL image into a new PIL image for enhanced export. Reading an
+    enhanced picture with PIL and save it directly does not work, so a new set of
+    pictures for each frame needs to be generated.
+
+    Parameters
+    ----------
+    pil_image:
+        the PIL image currently handled
+
+    Returns
+    -------
+    list:
+        a list of PIL images, one for each frame of the original PIL image
+    """
+    pil_image.seek(0)
+    images = [Image.fromarray(numpy.array(pil_image))]
+    pil_image.seek(1)
+    images.append(Image.fromarray(numpy.array(pil_image)))
+    pil_image.seek(2)
+    images.append(Image.fromarray(numpy.array(pil_image)))
+    return images
+
+
+def save_tif_stripped(pil_image, data, metadata):
+    """
+    Convert the existing pil image into a new TIF picture which can be used for
+    generating the required data for setting the payload.
+
+    Parameters
+    ----------
+
+    pil_image:
+        the PIL image currently handled
+    data:
+        the dictionary holding the data for the payload
+    metadata:
+        the JSON string holding the enhanced picture metadata
+
+    Returns
+    -------
+    data:
+        the updated dictionary holding the data for the payload
+    """
+    buff = io.BytesIO()
+    new_pil_images = create_new_pil_image(pil_image)
+    tiffinfo_dir = {TIFFTAG_IMAGEDESCRIPTION: metadata}
+    new_pil_images[0].save(
+        buff,
+        "TIFF",
+        compression="deflate",
+        save_all=True,
+        append_images=[new_pil_images[1], new_pil_images[2]],
+        tiffinfo=tiffinfo_dir,
+    )
+    buff.seek(0)
+    data["file_data"] = buff.read()
+    data["format"] = "tif"
+    buff.close()
+    return data
+
+
+def PIL_image_to_data(img, guid=None):
+    """
+    Convert the input image to a dictionary holding the data for the payload.
+
+    Parameters
+    ----------
+    img:
+        the input picture. It may be bytes or the path to the file to read
+    guid:
+        the guid of the image if it is an already available Qt image
+
+    Returns
+    -------
+    data:
+        A dictionary holding the data for the payload
+    """
+    imgbytes = img
+    if isinstance(img, str):
+        imgbytes = open(img, "rb")
+    data = {}
+    image = Image.open(imgbytes)
+    data["format"] = image.format.lower()
+    if data["format"] == "tiff":
+        data["format"] = "tif"
+    data["width"] = image.width
+    data["height"] = image.height
+    metadata = is_enhanced(image)
+    if metadata:
+        data = save_tif_stripped(image, data, metadata)
+        imgbytes.close()
+    else:
+        buff = io.BytesIO()
+        image.save(buff, "PNG")
+        buff.seek(0)
+        data["file_data"] = buff.read()
+        buff.close()
+        imgbytes.close()
+    return data
 
 
-def enve_image_to_data(img, guid=None):
```

Comment:
@mariostieriansys you've changed the name of this method - which is fine, but this is called by the EnSight-ADR interface via pydynamicreporting, so we need to update the version number of pydynamicreporting for this change so we don't break EnSight's build.

Reply:
I can change back the name if you prefer.
The nightly builds have now pinned the version of pydynamicreporting to 0.5.0, they are not getting the dev builds. So the change will be needed when we will work again on develop

Reply:
I forgot we had pinned down the version for now - so it's fine, we will just need to remember the methods name changes when we update. Thanks